src/extensions/yfm/Math/commands.ts

```diff
@@ -28,14 +26,30 @@ export const removeEmptyMathInlineIfCursorIsAtBeginning: Command = ({tr, schema}
     return false;
 };
 
-/**
- * If MathInline is before cursor – move cursor to the end of the math
- */
-export const moveCursorToEndOfMathInline: Command = ({tr, schema}, dispatch) => {
-    const $cursor = get$Cursor(tr.selection);
-    if ($cursor?.nodeBefore?.type === mathIType(schema)) {
-        dispatch?.(tr.setSelection(TextSelection.create(tr.doc, $cursor.pos - 1)));
-        return true;
-    }
-    return false;
+// Handle cursor movement at the boundary of a MathInline block
+export const handleArrowKey = (direction: 'right' | 'left'): Command => {
+    return ({tr, schema}, dispatch) => {
+        const $cursor = get$Cursor(tr.selection);
+
+        if ($cursor) {
+            const moveRight = direction === 'right';
+
+            const mathType = mathIType(schema);
+            const parentNodeType = $cursor.parent.type;
+            const nextNodeType = moveRight ? $cursor.nodeAfter?.type : $cursor.nodeBefore?.type;
+
+            const isAtBoundary =
+                $cursor.parentOffset === (moveRight ? $cursor.parent.content.size : 0);
+            const shouldMove =
+                nextNodeType === mathType || (isAtBoundary && parentNodeType === mathType);
+
+            if (shouldMove && dispatch) {
+                const newPos = $cursor.pos + (moveRight ? 1 : -1);
+                dispatch(tr.setSelection(TextSelection.create(tr.doc, newPos)));
+                return true;
+            }
```

Comment:
This code handles two actions at once. According to the first of the SOLID principles, plz split it into two separate functions